kotlinx-coroutines-core/common/src/channels/Produce.kt

```diff
@@ -141,3 +152,20 @@ internal open class ProducerCoroutine<E>(
         if (!processed && !handled) handleCoroutineException(context, cause)
     }
 }
+
+private class DropWhileBusyProducerCoroutine<E>(
+    parentContext: CoroutineContext,
+    channel: Channel<E>
+) : ProducerCoroutine<E>(parentContext, channel) {
+
+    private var hasSentFirstValue: Boolean = false
+
+    override suspend fun send(element: E) {
+        if (hasSentFirstValue) {
+            offer(element)
+        } else {
+            super.send(element)
+            hasSentFirstValue = true
+        }
+    }
+}
```

Comment:
Performance nit: Using `_channel.offer` and `_channel.send` would be slightly better here.